lib/src/navbar_router.dart

```diff
@@ -354,6 +369,10 @@ class _NavbarRouterState extends State<NavbarRouter>
                             if (widget.onCurrentTabClicked != null) {
                               widget.onCurrentTabClicked!();
                             }
+                            if (widget.hideBadgeOnPageChanged) {
```

Comment:
I think this invocation should be in the else part of 

```dart
if (NavbarNotifier.currentIndex == x) {
```

The badge should hide when user switches to a tab that contains a badge.

e.g if we have 4 tabs and we are on tab 0 and if Tab 1 contains a badge, switching to tab1 should hide the badge lmkwyt.

Reply:
Hello, but I have tested it and it worked as intended. Also, they can hide the badge on their current tab by tapping on it again too. I don't think theres any reason to change it 😅

Reply:
Okay, I tested the app its working as intended, But may I know how are the badges hiding by itself if its running the else part of the code?